Makefile

```diff
@@ -1,25 +1,56 @@
 PKG := resin
-VERSION := $(shell git describe --abbrev=0 --tags --dirty)
+VERSION := $(shell git describe --abbrev=0 --tags)
 EXECUTABLE := sshproxy
+RELEASE_TARBALLS := build/$(EXECUTABLE)-$(VERSION)_linux_arm5.tar.gz \
+ build/$(EXECUTABLE)-$(VERSION)_linux_arm7.tar.gz \
+ build/$(EXECUTABLE)-$(VERSION)_darwin_386.tar.gz \
+ build/$(EXECUTABLE)-$(VERSION)_linux_386.tar.gz \
+ build/$(EXECUTABLE)-$(VERSION)_darwin_amd64.tar.gz \
+ build/$(EXECUTABLE)-$(VERSION)_freebsd_amd64.tar.gz \
+ build/$(EXECUTABLE)-$(VERSION)_linux_amd64.tar.gz
+USER := resin-io
+REPO := sshproxy
 
 all: bin/$(EXECUTABLE)
 
-bin/$(EXECUTABLE):
-	go build -o "$@" ./$(PKG)
+dep:
+	go get -v ./...
 
-release: release/$(EXECUTABLE)-$(VERSION)_linux_arm5.tar.bz2 \
- release/$(EXECUTABLE)-$(VERSION)_linux_arm7.tar.bz2 \
- release/$(EXECUTABLE)-$(VERSION)_darwin_386.tar.bz2 \
- release/$(EXECUTABLE)-$(VERSION)_linux_386.tar.bz2 \
- release/$(EXECUTABLE)-$(VERSION)_darwin_amd64.tar.bz2 \
- release/$(EXECUTABLE)-$(VERSION)_freebsd_amd64.tar.bz2 \
- release/$(EXECUTABLE)-$(VERSION)_linux_amd64.tar.bz2
+bin/$(EXECUTABLE): dep
+	go build -o "$@" -v ./$(PKG)
 
-release-sign: release
-	for f in release/*.tar.bz2; do gpg --armor --detach-sign $$f; done
+lint-dep: dep
+	go get github.com/golang/lint/golint
+	go get golang.org/x/tools/cmd/goimports
+
+lint: lint-dep
+	goimports -d .
+	gofmt -e -l -s .
+	golint -set_exit_status ./...
+	go tool vet .
+
+test-dep: dep
+	go test -i -v ./...
+
+test: test-dep
+	go test -v ./...
+
+release: $(RELEASE_TARBALLS)
+
+upload-dep:
+	go get github.com/aktau/github-release
+
+upload: lint test release upload-dep
+	@test -n "${GITHUB_TOKEN}"
```

Comment:
Got it 👍 By the way it'd still be nice to have signed artifacts (possibly as part of the release notes?)